Assets/Scripts/Augment/GunFactory.cs

```diff
@@ -13,6 +13,24 @@ public static GameObject InstantiateGun(Item bodyPrefab, Item barrelPrefab, Item
         controller.Body = bodyPrefab;
         controller.Barrel = barrelPrefab;
         controller.Extension = extensionPrefab;
+        
+
+        // Initialize everything
+        gun.GetComponent<GunFactory>().InitializeGun();
+
+        return gun;
+    }
+
+    public static GameObject InstantiateGun(Item bodyPrefab, Item barrelPrefab, Item extensionPrefab, Transform parent, Vector3 scale)
```

Comment:
The overload has been removed and GallerMenu.cs now sets the parent and scale itself instead.

Reply:
oh
oh hold on
I didn't see the extra parameters
um
yeah okay though that's some functionality that is nice to have
maybe do an overload but write it so that it calls the other InstantiateGun instead of duplicating the code :/

Reply:
According to @SondreHus, the overload was unnecessary. So I'm opting to keep it as it is now.